apps/condo/domains/common/hooks/useBroadcastChannel.tsx

```diff
@@ -0,0 +1,45 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+
+type UseBroadcastChannelOptions = {
+    sendInCurrentTab?: boolean
+}
+
+const DEFAULT_OPTIONS: UseBroadcastChannelOptions = {
+    sendInCurrentTab: true,
+}
+
+export function useBroadcastChannel<T> (
+    channelName: string,
+    onMessageReceived: (message: T) => void,
+    options: UseBroadcastChannelOptions = DEFAULT_OPTIONS
+) {
+    const { sendInCurrentTab } = options
+
+    const messageSender = useRef<BroadcastChannel | null>(null)
+    const messageReceiver = useRef<BroadcastChannel | null>(null)
+
+    useEffect(() => {
+        messageSender.current = new BroadcastChannel(channelName)
+        messageReceiver.current = sendInCurrentTab ? new BroadcastChannel(channelName) : messageSender.current
+
+        messageReceiver.current.onmessage = (event) => onMessageReceived(event.data)
+    }, [channelName, onMessageReceived, sendInCurrentTab])
```

Comment:
You should close the channel if you want to create a new one or terminate a connection